lightning/src/ln/msgs.rs

```diff
@@ -74,63 +74,107 @@ pub struct Init {
+	pub data: String,
 }
 
 /// A ping message to be sent or received from a peer
 pub struct Ping {
-	pub(crate) ponglen: u16,
-	pub(crate) byteslen: u16,
+	/// The desired response length
+	pub ponglen: u16,
+	/// The ping packet size.
+	/// This field is not sent on the wire. byteslen zeros are sent.
+	pub byteslen: u16,
 }
 
 /// A pong message to be sent or received from a peer
 pub struct Pong {
-	pub(crate) byteslen: u16,
+	/// The pong packet size.
+	/// This field is not sent on the wire. byteslen zeros are sent.
+	pub byteslen: u16,
 }
 
 /// An open_channel message to be sent or received from a peer
 #[derive(Clone)]
 pub struct OpenChannel {
-	pub(crate) chain_hash: BlockHash,
-	pub(crate) temporary_channel_id: [u8; 32],
-	pub(crate) funding_satoshis: u64,
-	pub(crate) push_msat: u64,
-	pub(crate) dust_limit_satoshis: u64,
-	pub(crate) max_htlc_value_in_flight_msat: u64,
-	pub(crate) channel_reserve_satoshis: u64,
-	pub(crate) htlc_minimum_msat: u64,
-	pub(crate) feerate_per_kw: u32,
-	pub(crate) to_self_delay: u16,
-	pub(crate) max_accepted_htlcs: u16,
-	pub(crate) funding_pubkey: PublicKey,
-	pub(crate) revocation_basepoint: PublicKey,
-	pub(crate) payment_point: PublicKey,
-	pub(crate) delayed_payment_basepoint: PublicKey,
-	pub(crate) htlc_basepoint: PublicKey,
-	pub(crate) first_per_commitment_point: PublicKey,
-	pub(crate) channel_flags: u8,
-	pub(crate) shutdown_scriptpubkey: OptionalField<Script>,
+	/// The genesis hash of the blockchain where the channel is to be opened
+	pub chain_hash: BlockHash,
+	/// A temporary channel ID, until the funding is established
+	pub temporary_channel_id: [u8; 32],
+	/// The channel value
+	pub funding_satoshis: u64,
+	/// The amount to push to the counterparty as part of the open, in milli-satoshi
+	pub push_msat: u64,
+	/// The threshold below which outputs should be omitted
+	pub dust_limit_satoshis: u64,
+	/// The maximum total HTLC value in flight, in milli-satoshi
+	pub max_htlc_value_in_flight_msat: u64,
+	/// The minimum value for the counterparty to keep in the channel
+	pub channel_reserve_satoshis: u64,
+	/// The minimum HTLC size, in milli-satoshi
+	pub htlc_minimum_msat: u64,
+	/// The feerate per 1000-weight of the transaction
+	pub feerate_per_kw: u32,
+	/// The minimum to_self_delay for the counterparty's transactions
+	pub to_self_delay: u16,
+	/// The maximum number of accepted HTLCs
+	pub max_accepted_htlcs: u16,
+	/// The funding public key
+	pub funding_pubkey: PublicKey,
+	/// The funding public key
+	pub revocation_basepoint: PublicKey,
+	/// The payment point
+	pub payment_point: PublicKey,
+	/// The delayed payment basepoint
+	pub delayed_payment_basepoint: PublicKey,
+	/// The HTLC basepoint
+	pub htlc_basepoint: PublicKey,
+	/// The first transaction's per commitment point
+	pub first_per_commitment_point: PublicKey,
+	/// Channel flags
+	pub channel_flags: u8,
+	/// Optionally, the scriptPubkey when we collaboratively close
```

Comment:
I would say this is a "request to pre-set" rather than a commitment, since the counterparty may fail to enforce this without any negative consequences.